[Mind-21] 매칭 요청 기능 구현

TaskerJang opened the pull request:
# Pull Request

## 💡 PR 요약
매칭 요청 기능을 구현했습니다. 사용자가 전문가에게 매칭을 요청할 수 있는 API를 개발하였습니다.

## 🔍 주요 변경사항
- 매칭 엔티티 확장 및 매칭 상태 열거형 추가
- 매칭 요청/응답 DTO 클래스 구현
- 매칭 요청 API 컨트롤러 및 서비스 로직 구현
- 매칭 요청 기능 단위 테스트 코드 작성
- BaseResponseStatus에 매칭 관련 상태 추가

## 🔗 연관된 이슈
close #21

## ✅ 체크리스트
- [x] 테스트 코드를 작성하였나요?
- [x] 관련 문서를 업데이트하였나요?
- [ ] Breaking Change가 있나요?
- [x] 코드 포맷팅과 린트 검사를 완료하였나요?

## 🙏 리뷰어 참고사항


## 📋 추가 컨텍스트
문제 유형에 따라 매칭이 되도록 하는것을 깜빡했네요.. 그것까지 해서 보완해야할것 같습니다 ㅠㅠ

Commit:
[feat]: 매칭 엔티티 및 상태 클래스 구현
- Matching 엔티티 확장
- MatchingStatus 열거형 추가

Commit:
[feat]: 매칭 요청/응답 DTO 클래스 구현
- MatchingRequestDto 구현
- MatchingResponseDto 구현

Commit:
[feat]: 매칭 저장소 및 서비스 구현
- MatchingRepository 인터페이스 구현
- MatchingService 비즈니스 로직 구현

Commit:
[feat]: 매칭 요청 API 컨트롤러 구현
- MatchingController 구현
- BaseResponseStatus에 매칭 관련 상태 추가

Commit:
[test]: 매칭 요청 기능 테스트 코드 추가 (#21)
- MatchingServiceTest 클래스 구현
- 매칭 요청 성공 케이스 테스트
- 전문가 존재하지 않을 경우 실패 케이스 테스트
- 매칭 상태 기본값 검증 테스트

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/dto/MatchingResponseDto.java:
```diff
@@ -0,0 +1,34 @@
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class MatchingResponseDto {
```
모든 dto 클래스 네이밍에 Dto가 들어가있습니다!! 저희 규칙에서는 ~~Request , ~~Response로 통일하기로 했었던걸로 알고있습니다.!! 컨벤션을 지켜주시면 감사할것 같습니다

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/service/MatchingService.java:
```diff
@@ -1 +1,41 @@
+    @Transactional
+    public MatchingResponseDto requestMatching(Long userId, MatchingRequestDto requestDto) {
+        // 전문가 존재 여부 확인
+        Expert expert = expertRepository.findById(requestDto.getExpertId())
```
해당부분 메서드로 분리하면 좋을것 같습니다!

hwan2-99 commented on src/main/java/com/mindgoal/domain/matching/service/MatchingService.java:
```diff
@@ -1 +1,41 @@
+                .build();
+
+        // 매칭 저장
+        Matching savedMatching = matchingRepository.save(matching);
```
해당 로직이 필요할까요??
마무리를 
MachingResponseDto.from(matchingRepository.save(matching));
로변경하면 더 가독성이 높아질 것 같습니다!

hwan2-99 commented on src/test/java/com/mindgoal/backend/matching/service/MatchingServiceTest.java:
```diff
@@ -0,0 +1,155 @@
+    private UserRepository userRepository;
+
+    @BeforeEach
+    void setUp() {
```
해당부분은 제가 어노테이션으로 각 테스트 후에 모든 데이터를 제거하는걸로 구현해뒀습니다.!! 해당부분이 있으면 불필요한 테스트덕에 전체테스트가 실패할 수도 있어 삭제해주시면 좋을것 같습니다!

hwan2-99 reviewed (commented):
고생하셨습니다.!

Commit:
[refactor]: 매칭 관련 클래스 네이밍 컨벤션 적용

- MatchingController, MatchingRequest, MatchingResponse 컨벤션 적용
- MatchingService 코드 리팩토링
- MatchingServiceTest의 불필요한 setUp 메소드 제거
